### PR TITLE
Transform impls

### PR DESCRIPTION
**Pull request by Knize**

### CODE REVIEW

File: j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/jk.kt [hunk 1]
```diff
@@ -49,104 +49,153 @@ class JKClassImpl(override var modifierList: JKModifierList, override var name:
...
+
+    override fun <D> transformChildren(transformer: JKTransformer<D>, data: D) {
+        expression = expression?.transform(transformer, data)
+        operator = operator.transform(transformer, data)
+    }
 }
 
-class JKPostfixExpressionImpl(override val expression: JKExpression,
-                              override val operator: JKOperatorIdentifier) : JKPostfixExpression, JKElementBase() {
+class JKPostfixExpressionImpl(override var expression: JKExpression?,
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/jk.kt]: Why made `expression` nullable?

Comment (semoro) [reply j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/jk.kt]: OK, after discussion with @Dimach, we decided, that it shouldn't be nullable here, and in `JKPostfixExpression`, but should to in `JKUnaryExpression`

**Review by semoro — CHANGES_REQUESTED** [review]

**Comment by Knize** [comment]
OK

File: j2k/newSrc/org/jetbrains/kotlin/j2k/tree/jk.kt [hunk 2]
```diff
@@ -43,14 +43,14 @@ interface JKStatement : JKElement
 interface JKExpression : JKStatement
 
 interface JKBinaryExpression : JKExpression {
-    val left: JKExpression
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/tree/jk.kt]: Why did interfaces change?

File: j2k/newSrc/org/jetbrains/kotlin/j2k/tree/j.kt [hunk 3]
```diff
@@ -54,8 +54,8 @@ interface JKJavaFieldAccessExpression : JKFieldAccessExpression {
 }
 
 interface JKJavaNewExpression : JKExpression {
-    val identifier: JKClassReference
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/tree/j.kt]: Why those `val` became `var`? And other in this file

File: j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/jk.kt [hunk 4]
```diff
@@ -49,104 +49,153 @@ class JKClassImpl(override var modifierList: JKModifierList, override var name:
 }
 
 
-class JKNameIdentifierImpl(override val name: String) : JKNameIdentifier, JKElementBase() {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/jk.kt:52]: This shouldn't be mutable

File: j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/j.kt [hunk 5]
```diff
@@ -48,7 +48,7 @@ class JKJavaStringLiteralExpressionImpl(override val text: String) : JKJavaStrin
 }
 
 
-class JKJavaAccessModifierImpl(override val type: JKJavaAccessModifier.AccessModifierType) : JKJavaAccessModifier, JKElementBase() {
```
Comment (semoro) [inline j2k/newSrc/org/jetbrains/kotlin/j2k/tree/impl/j.kt]: This shouldn't be mutable

**Review by semoro — CHANGES_REQUESTED** [review]

**Review by semoro — APPROVED** [review]

**Pull request merged**